src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs

```diff
@@ -1056,7 +1060,19 @@ protected virtual void PrepareContainerForItemOverride(DependencyObject element,
 					// Set the datacontext first, then the binding.
 					// This avoids the inner content to go through a partial content being
 					// the result of the fallback value of the binding set below.
-					containerAsContentControl.DataContext = item;
+
+					var displayMemberPath = DisplayMemberPath;
+					if (string.IsNullOrEmpty(displayMemberPath))
```

Comment:
You should make this available to the `containerAsContentPresenter` code path as well - this is hit when using a bare `<ItemsControl \>`.